### PR TITLE
Reduce Metrics/MethodLength Max to 14

### PR DESCRIPTION
**Pull request by lumeet**
Several method extractions and some minor cleanup.
-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/runner.rb [hunk 1]
```diff
@@ -216,6 +200,29 @@ def do_inspection_loop(file, processed_source)
...
+
+      # It is also possible for a cop to keep adding indefinitely to a file,
+      # making it bigger and bigger. If the inspection loop runs for an
+      # excessively high number of iterations, this is likely happening.
+      iterations = 0
+
+      loop do
+        check_for_infinite_loop(source, offenses)
+
+        if (iterations += 1) > 200
```
Comment (bbatsov) [inline lib/rubocop/runner.rb]: 200 should probably be made a constant.

File: lib/rubocop/cop/style/space_inside_string_interpolation.rb [hunk 2]
```diff
@@ -19,25 +19,39 @@ class SpaceInsideStringInterpolation < Cop
...
 
         def on_dstr(node)
+          each_style_violation(node) do |final_node, msg|
+            add_offense(final_node, :expression, msg)
+          end
+        end
+
+        private
+
+        def each_style_violation(node)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/space_inside_string_interpolation.rb]: I feel such names are a bit misleading, as this method is not really iterating through a collection with `each_something` often implies. I know you've already used them at some places, but I thought it makes sense to mention this.

Comment (lumeet) [inline lib/rubocop/cop/style/space_inside_string_interpolation.rb]: Fair enough. I tend to use `each_` to tell that you can expect many of something. If it's only naming, what would you prefer instead?

File: lib/rubocop/runner.rb (hunk 1 above)
Comment (lumeet) [inline lib/rubocop/runner.rb]: Ouch, I already did that once but rewrote the whole thing and forgot it.

**Comment by rrosenblum** [comment]
Out of curiosity, is there an end goal number for the Metrics improvements that have been going on?

File: lib/rubocop/cop/style/space_inside_string_interpolation.rb (hunk 2 above)
Comment (bbatsov) [inline lib/rubocop/cop/style/space_inside_string_interpolation.rb]: Ah, I see your point. Guess this makes sense as well. I'd probably use the name `on_style_violation`, though. I think we can figure this out later...

**Comment by lumeet** [comment]
@rrosenblum Personally I've kept the RuboCop defaults in `config/default.yml` as the targets but I don't know if we're ever going to reach them. While interesting, it takes increasingly more time and gives less benefit as we're getting closer.

**Pull request merged**